app/assets/javascripts/admin/orders/controllers/orders_controller.js.coffee

```diff
@@ -16,11 +16,41 @@ angular.module("admin.orders").controller "ordersCtrl", ($scope, $timeout, Reque
   $scope.rowStatus = {}
 
   $scope.initialise = ->
-    $scope.per_page = 15
-    $scope.q = {
-      completed_at_not_null: true
-    }
-    $scope.fetchResults()
+    params = $location.search()
+    # Get all params from url (if presents) and then put into $scope
+    $scope.per_page = parseInt(params.per_page) || 15
+    if (params.shipping_method_id)
+      $scope.shipping_method_id = parseInt(params.shipping_method_id)
+    
+    # All params specified in the query 'q'
+    $scope.q = {}
+    ["completed_at_gteq",
+     "completed_at_lteq",
+     "state_eq",
+     "number_cont",
+     "email_cont"
+     "bill_address_firstname_start",
+     "bill_address_lastname_start"
+     ]
+    .map((paramKey) -> 
```

Comment:
`map` is used to map all items of a list to different items of a new list. You are not using the new list though. You are iterating over the items of the list. Coffeescript provides a syntax for that: https://coffeescript.org/#loops

This is not prescriptive but maybe I would write it like this (untested):
```coffee
searchKeys = [
     "completed_at_gteq",
     "completed_at_lteq",
     "state_eq",
     "number_cont",
     "email_cont"
     "bill_address_firstname_start",
     "bill_address_lastname_start"
]
for paramKey in searchKeys when params["q["+paramKey+"]"]
  $scope.q[paramKey] = params["q["+paramKey+"]"]
```

Reply:
I've the habit (like many others ;)) to use map as an iterator over list (and use it to run a callback/lambda over each item).

Reply:
Modified!